spec/lib/stylesheet/manager_spec.rb

```diff
@@ -872,6 +872,67 @@ def write(x)
       content = StylesheetCache.last.content
       expect(content).to match(/# sourceMappingURL=[^\/]+\.css\.map\?__ws=test\.localhost/)
     end
+
+    it "generates precompiled CSS with a missing upload" do
+      image = file_from_fixtures("logo.png")
+      upload = UploadCreator.new(image, "logo.png").create_for(-1)
+
+      scheme = ColorScheme.create!(name: "scheme")
+      core_targets = [:desktop, :mobile, :desktop_rtl, :mobile_rtl, :admin, :wizard]
+      theme_targets = [:desktop_theme, :mobile_theme]
+
+      default_theme = Fabricate(:theme, color_scheme: scheme).tap do |t|
+        field = ThemeField.create!(
+          theme_id: t.id,
+          target_id: Theme.targets[:common],
+          name: "logo",
+          value: "",
+          upload_id: upload.id,
+          type_id: ThemeField.types[:theme_upload_var]
+        )
+
+        t.set_field(
+          target: :common,
+          name: :scss,
+          value: "body { background: url($logo); }"
```

Comment:
```suggestion
          value: "body { background: url($logo); border: 3px solid green; }"
```

Small suggestion, I think it's useful to add a test CSS rule and then in the spec check that it is still there in the compiled CSS output. That way we have fidelity that the only missing thing will be the SCSS rule with missing upload.